### DISCUSSION TITLE
Where is GM Screen Data Stored?

### DISCUSSION THREAD

**Discussion by TheErikkuBlade** [angometry/trinium-chat-buttons#16] [Q&A] [answered]
Hi, I'm a GM for multiple games and I also am tech support/module manager for one of my other GMs and I wanted to quickly get a GM Screen setup for them. I'm curious on how the GM Screen data is stored. Does it use cookies? Is it world based or client based? Is there some way to export/import changes made from one world's GM Screen to another world?

**Comment by angometry** [comment, marked as answer]
Hi!
Currently the content of each tab is stored quite simply as a string in the game's world settings:

https://github.com/angometry/trinium-chat-buttons/blob/main/src/module/settings.js#L23
https://github.com/angometry/trinium-chat-buttons/blob/main/src/module/settings.js#L161

Foundry stores these settings in a database and I never looked into how that whole thing works, but I do know a module Forien's Copy Environment exists, which allows migratng settings. Since it's a simple string, it should probably handle such a migration with no issues.

This module was a bit of a fun experiment to see what could be done, so it's all a bit hacky. Once I'm done with some important work I plan to refactor it completely and already have ideas on how to make contents export/importable and stored in a more appropriate format.
ETA on when I can get working is probably a week or two, so you might want to wait for it before going too deep into configuration.

**Reply by TheErikkuBlade** [reply]
Sounds good! I wish you luck with that and thanks for getting back to me! :)
Have a great day!
